app/react/Library/Library.js

```diff
@@ -57,9 +57,15 @@ export default class Library extends RouteHandler {
 
   render() {
     return (
-      <LibraryLayout>
-        <LibraryModeToggleButtons storeKey="library" zoomIn={this.zoomIn} zoomOut={this.zoomOut} />
-        <DocumentsList storeKey="library" />
+      <LibraryLayout fixedSidePanels={this.props.fixedSidePanels}>
+        <LibraryModeToggleButtons
+          storeKey="library"
+          zoomIn={this.zoomIn}
+          zoomOut={this.zoomOut}
+          showColumnSelector={this.props.viewer !== undefined}
```

Comment:
I don't like this approach for the entire adding of the TableViewer. I commented on the side panel "fixed" thing, and here we find the same criteria: "If I passed the viewer prop, show the column selector".  What if we later on add a new viewer that is named "mapViewer", or "relationshipsTreeViewer", all of  a sudden we get a column selector?

Please, do this very explicitly! Something like "if props.viewer is an instance of TableView" or, for example, explicitly passing a "showColumnSelector" prop that is controlled where the viewer is defined.  Something really explicit that we want this shown, instead of a secondary effect of passing a viewer.

Reply:
Yeah, we overlooked this. Will correct it.